packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-feedback-modal.tsx

```diff
@@ -0,0 +1,243 @@
+      this.smartSnippet.openFeedbackModal();
+      this.currentAnswer = undefined;
+    }
+  }
+
+  initialize() {
+    this.smartSnippet = buildSmartSnippet(this.bindings.engine);
+  }
+
+  private sendFeedback() {
+    if (this.currentAnswer === 'other') {
+      this.smartSnippet.sendDetailedFeedback(this.detailsInputRef!.value); // TODO: Add textarea
+    } else {
+      this.smartSnippet.sendFeedback(this.currentAnswer!);
+    }
+    this.isOpen = false;
+  }
+
+  private close() {
+    this.isOpen = false;
+    this.smartSnippet.closeFeedbackModal();
+  }
+
+  private renderHeader() {
+    return (
+      <h1 slot="header">
+        {this.bindings.i18n.t('smart-snippet-feedback-explain-why')}
+      </h1>
+    );
+  }
+
+  private renderOptions() {
+    const options: {
+      id: string;
+      label: string;
+      isChecked: boolean;
+      onCheck: () => void;
+    }[] = [
+      {
+        id: 'does-not-answer',
+        label: this.bindings.i18n.t(
+          'smart-snippet-feedback-reason-does-not-answer'
+        ),
+        isChecked: this.currentAnswer === 'does_not_answer',
+        onCheck: () => (this.currentAnswer = 'does_not_answer'),
+      },
+      {
+        id: 'partially-answers',
+        label: this.bindings.i18n.t(
+          'smart-snippet-feedback-reason-partially-answers'
+        ),
+        isChecked: this.currentAnswer === 'partially_answers',
+        onCheck: () => (this.currentAnswer = 'partially_answers'),
+      },
+      {
+        id: 'was-not-a-question',
+        label: this.bindings.i18n.t(
+          'smart-snippet-feedback-reason-was-not-a-question'
+        ),
+        isChecked: this.currentAnswer === 'was_not_a_question',
+        onCheck: () => (this.currentAnswer = 'was_not_a_question'),
+      },
+      {
+        id: 'other',
+        label: this.bindings.i18n.t('smart-snippet-feedback-reason-other'),
+        isChecked: this.currentAnswer === 'other',
+        onCheck: () => (this.currentAnswer = 'other'),
+      },
+    ];
+
+    return (
+      <fieldset>
+        <legend part="reason-title" class="font-bold text-on-background">
+          {this.bindings.i18n.t('smart-snippet-feedback-select-reason')}
+        </legend>
+        {options.map(({id, label, isChecked, onCheck}) => (
+          <div key={id} class="text-base leading-4 text-neutral-dark mt-2">
+            <input
+              part="reason-radio"
+              type="radio"
+              name="answer"
+              id={id}
+              checked={isChecked}
+              onChange={(e) =>
+                (e.currentTarget as HTMLInputElement | null)?.checked &&
+                onCheck()
+              }
+              class="mr-2 w-4 h-4"
+              required
+            />
+            <label part="reason-label" htmlFor={id}>
+              {label}
+            </label>
+          </div>
+        ))}
+      </fieldset>
+    );
+  }
+
+  private renderDetails() {
+    if (this.currentAnswer !== 'other') {
```

Comment:
personal preference, I think when there is only one option (as in, we will only render for the 'other' option), it's easier to understand if you have the affirmative and then the render inside, like
```
if (this.currentAnswer === 'other') {
  return <component>
}
return null
```

Reply:
I also generally prefer the affirmative first, but when one case returns a falsy/null value, I prefer to put that one first. I'll keep the current pattern because that's what I've written in most places, but I'm not against your pattern either.